controller.js

```diff
@@ -24,17 +24,24 @@ const getAllArticles = (request, response, next) => {
     });
 };
 
-//This function checks if an article exists before returning either the article or an error message
+//This function firstly checks the request parameter is a number.
+//It then passes the parameter to the model, and on return, checks if an article exists before returning either the article or an error message
 const getArticleById = (request, response, next) => {
   const article_id = request.params.article_id;
-  fetchArticleById(article_id)
-    .then((article) => {
-      if (article) response.status(200).send({ article });
-      else response.status(404).send({ msg: "Article not found" });
-    })
-    .catch((error) => {
-      next(error);
-    });
+  if (isNaN(article_id)) {
```

Comment:
Again when it comes to error use promise rejects which you can then build error handlers for instead of sending errors as responses.